src/main/java/com/example/demo/assembler/UserAssembler.java

```diff
@@ -0,0 +1,28 @@
+package com.example.demo.assembler;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.example.demo.model.User;
+import com.example.demo.response.dto.UserDto;
+
+@Component
+public class UserAssembler {
+	
+	@Autowired
+	UserDto userDto;
+	
+	public UserDto toModel(Optional<User> user)
+	{
```

Comment:
Don't take Optional in the input. This code will throw nullpointerexception @Line number 21 if user not found in optional<User>. Handle Optional in service layer returns the perfect object to controller or assembler